Remotion/Web/Development.WebTesting.IntegrationTests/Accessibility/AccessibilityAnalyzerTest.cs

```diff
@@ -0,0 +1,114 @@
+﻿// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.Linq;
+using NUnit.Framework;
+using Remotion.Web.Development.WebTesting.Accessibility;
+using Remotion.Web.Development.WebTesting.ControlSelection;
+using Remotion.Web.Development.WebTesting.PageObjects;
+using Remotion.Web.Development.WebTesting.WebFormsControlObjects;
+using Remotion.Web.Development.WebTesting.WebFormsControlObjects.Selectors;
+
+namespace Remotion.Web.Development.WebTesting.IntegrationTests.Accessibility
+{
+  [TestFixture]
+  public class AccessibilityAnalyzerTest : IntegrationTest
+  {
+    [Test]
+    public void AnalyzeIgnoreCssSelector ()
+    {
+      Start<HtmlPageObject> ("Accessibility/FormElementWithoutLabel.html");
+      var analyzer = Helper.CreateAccessibilityAnalyzer();
+
+      analyzer.IgnoreCssSelector ("#input");
+      var result = analyzer.Analyze();
+
+      Assert.That (result.Violations, Is.Empty);
+    }
+
+    [Test]
+    public void AnalyzePageObject ()
+    {
+      var home = Start<HtmlPageObject> ("Accessibility/FormElementWithoutLabel.html");
+      var analyzer = Helper.CreateAccessibilityAnalyzer();
+
+      var result = home.Analyze (analyzer);
+      var violation = result.Violations.Single();
+
+      Assert.That (violation.Rule.ID, Is.EqualTo (AccessibilityRuleID.Label));
+    }
+
+
+    [Test]
+    public void Analyze_ExcludeRule ()
```

Comment:
Hmm...I think I want to have a non-empty result at the end of the exclude tests. otherwise, we don't know if the exclude removed just the specified rules or all rules.

Reply:
Would this also be needed for multiple excludes/ignores?

Reply:
yeah, think so.